packages/common/src/ArcGISContext.ts

```diff
@@ -225,6 +248,40 @@ export interface IArcGISContext {
    * Refresh the current user, including their groups
    */
   refreshUser(): Promise<void>;
+
+  /**
+   * Add an entry to the user's history
+   * If not authenticated, this function will no-op
+   * @param entry
+   */
+  addToHistory(entry: IHubHistoryEntry): Promise<void>;
+
+  /**
+   * Remove a specific entry from the user's history
+   * @param entry
+   */
+  removeFromHistory(entry: IHubHistoryEntry): Promise<void>;
+
+  /**
+   * Simple clear of the all of the user's history
+   */
+  clearHistory(): Promise<void>;
+
+  /**
+   * Check specific permission for the current user, and optionally an entity
+   * @param permission
+   * @param entity
+   */
+  checkPermission(
+    permission: Permission,
+    entity?: HubEntity
+  ): IPermissionAccessResponse;
```

Comment:
This will be convenient. I guess we should generally just call this method instead of importing checkPermission, calling it, and passing it context?

Reply:
sure - I'd not really considered that, but it's same-same under the hood :)